src/testing/testing-sys.ts

```diff
@@ -1,4 +1,4 @@
-import type { CompilerSystem } from '@stencil/core/internal';
+import type { CompilerSystem } from '@stencil/core/declarations';
```

Comment:
Needed to change this import since after using the `TestingSystem` in the compiler sub-module, we ran into a circular dependency that (I think) was the culprit of type errors. This change resolved those issues, but generated invalid code in the bundled output, so I had to change the `scripts/bundles/testing` script to account for that.

The alternative to all this is that we move the `TestingSystem` type into our `src/declarations` module, but that felt kinda weird since this type is specific to testing.

Reply:
I'm not entirely sure the full context of the errors you were running into here, so this may not be helpful at all 🙃 We export `createTestingSystem` from the alias `@stencil/core/testing` as `mockCompilerSystem`. I was able to revert the changes to this file and [these changes](https://github.com/ionic-team/stencil/pull/4897/files#diff-c55d79f7b0f5bb6231fd6e1d7f9d0c694cfead8d0478acd3141f357cc35f00ebR130-R142) in `scripts/bundles/testing` by using that instead (and subsequently changing the imports in the affected test files). I ran unit tests/karma tests locally, but I'm not sure how to best test this otherwise. Can you take a look at the patch attached here and let me know if something like this will work (`git apply node-20-dot-8-patch.patch`)

Patch file: [node-20-dot-8-patch.patch](https://github.com/ionic-team/stencil/files/13268767/node-20-dot-8-patch.patch)